hbase-server/src/test/java/org/apache/hadoop/hbase/regionserver/TestRegionInfoBuilder.java

```diff
@@ -302,30 +283,28 @@ public void testConvert() {
     HBaseProtos.RegionInfo info = HBaseProtos.RegionInfo.newBuilder()
       .setTableName(HBaseProtos.TableName.newBuilder()
         .setQualifier(UnsafeByteOperations.unsafeWrap(tableName.getQualifier()))
-        .setNamespace(UnsafeByteOperations.unsafeWrap(tableName.getNamespace()))
-        .build())
+        .setNamespace(UnsafeByteOperations.unsafeWrap(tableName.getNamespace())).build())
       .setStartKey(UnsafeByteOperations.unsafeWrap(startKey))
-      .setEndKey(UnsafeByteOperations.unsafeWrap(endKey))
-      .setSplit(split)
-      .setRegionId(regionId)
+      .setEndKey(UnsafeByteOperations.unsafeWrap(endKey)).setSplit(split).setRegionId(regionId)
       .build();
 
     convertedRi = ProtobufUtil.toRegionInfo(info);
-    RegionInfo expectedRi = RegionInfoBuilder.newBuilder(tableName)
-            .setStartKey(startKey)
-            .setEndKey(endKey)
-            .setSplit(split)
-            .setRegionId(regionId)
-            .setReplicaId(0).build();
+    RegionInfo expectedRi = RegionInfoBuilder.newBuilder(tableName).setStartKey(startKey)
+      .setEndKey(endKey).setSplit(split).setRegionId(regionId).setReplicaId(0).build();
 
     assertEquals(expectedRi, convertedRi);
   }
 
-  // Duplicated method in TestRegionInfoDisplay too.
-  private HRegionInfo convert(RegionInfo ri) {
-    HRegionInfo hri = new HRegionInfo(
-    ri.getTable(), ri.getStartKey(), ri.getEndKey(), ri.isSplit(), ri.getRegionId());
-    hri.setOffline(ri.isOffline());
-    return hri;
+  @Test
+  public void testNewBuilderWithRegionInfo() {
+    RegionInfo ri = RegionInfoBuilder.newBuilder(name.getTableName()).build();
+    RegionInfo ri2 = RegionInfoBuilder.newBuilder(ri).build();
+    assertEquals(ri, ri2);
+
+    // make sure that the region name and encoded name are changed, see HBASE-24500 for more
+    // details.
```

Comment:
Add more test case?

Reply:
For example? I think this is enough?

Reply:
As you said on the issue desc, set different startkey or endkey?